iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/NodeBreakerVoltageLevel.java

```diff
@@ -565,13 +555,24 @@ private Integer getEdge(String switchId, boolean throwException) {
     @Override
     public Iterable<Terminal> getTerminals() {
         return FluentIterable.from(graph.getVerticesObj())
-                .filter(Predicates.notNull())
+                .filter(NodeBreakerVoltageLevel::isValid)
                 .transform(Functions.identity());
     }
 
     @Override
     public Stream<Terminal> getTerminalStream() {
-        return graph.getVertexObjectStream().filter(Objects::nonNull).map(Function.identity());
+        return graph.getVertexObjectStream()
+                .filter(NodeBreakerVoltageLevel::isValid)
+                .map(Function.identity());
+    }
+
+    private static NodeTerminal getTerminal(UndirectedGraph<NodeTerminal, SwitchImpl> graph, int node) {
+        NodeTerminal terminal = graph.getVertexObject(node);
+        return isValid(terminal) ? terminal : null;
+    }
+
+    private static boolean isValid(NodeTerminal nodeTerminal) {
```

Comment:
For this method and the getTerminal method, I think we should either rename them or explain in comments/javadoc the reason for considering terminals with no connectable "invalid".
Maybe we should document this design choice (having Terminals for all vertices, but not using them for most methods) in the header?